src/nexgen/nxs_utils/Axes.py

```diff
@@ -0,0 +1,49 @@
+"""
+Utilities for axes definition
+"""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Literal, Tuple
+
+from ..utils import Point3D
+
+
+# Define axes and scans
+@dataclass
+class Axis:
+    """
+    Define an axis object for goniometer or detector.
+    """
+
+    name: str
+    depends: str
```

Comment:
It still feels like this should be an `Axis`, we're basically creating a linked list